llvm/lib/Target/Xtensa/XtensaISelLowering.cpp

```diff
@@ -394,20 +382,18 @@ SDValue XtensaTargetLowering::LowerFormalArguments(
   }
 
   if (IsVarArg) {
-    static const MCPhysReg XtensaArgRegs[6] = {
-        Xtensa::A2, Xtensa::A3, Xtensa::A4, Xtensa::A5, Xtensa::A6, Xtensa::A7};
-    ArrayRef<MCPhysReg> ArgRegs = ArrayRef(XtensaArgRegs);
+    ArrayRef<MCPhysReg> ArgRegs = ArrayRef(IntRegs);
     unsigned Idx = CCInfo.getFirstUnallocated(ArgRegs);
```

Comment:
Directly passing IntRegs to getFirstUnallocated should work I think

Reply:
Fixed